docs/hazmat/primitives/asymmetric/serialization.rst

```diff
@@ -459,6 +459,187 @@ An example ECDSA key in OpenSSH format::
+        If a CA does not wish to number its certificates, it must set this
+        field to zero.
+
+    .. attribute:: type
+
+        :type: :class:`SSHCertificateType`
+
+        Type specifies whether this certificate is for identification of a user
+        or a host.
+
+    .. attribute:: key_id
+
+        :type: bytes
+
+        This is a free-form text field that is filled in by the CA at the time
+        of signing; the intention is that the contents of this field are used to
+        identify the identity principal in log messages.
+
+    .. attribute:: valid_principals
+
+        :type: list[bytes]
+
+        "valid principals" is a list containing zero or more principals as
+        byte strings. These principals list the names for which this
+        certificate is valid; hostnames for host certificates and
+        usernames for user certificates. As a special case, an
+        empty list means the certificate is valid for any principal of
+        the specified type.
+
+    .. attribute:: valid_before
+
+        :type: :class:`datetime.datetime`
+
+        A naïve datetime representing the UTC time before which the certificate
+        is valid. **This time is not inclusive.**
+
+    .. attribute:: valid_after
+
+        :type: :class:`datetime.datetime`
+
+        A naïve datetime representing the UTC time after which the certificate
+        is valid. **This time is inclusive.**
+
+    .. attribute:: critical_options
+
+        :type: dict[bytes, bytes]
+
+        Critical options is a dict of zero or more options that are
+        considered critical for the certificate to be considered valid. If
+        any of these options are not supported by the implementation, the
+        certificate must be rejected.
+
+    .. attribute:: extensions
+
+        :type: dict[bytes, bytes]
+
+        Extensions is a dict of zero or more options that are considered
+        non-critical for the certificate to be considered valid. If any of
+        these options are not supported by the implementation, the
+        implementation may safely ignore them.
+
+    .. method:: signature_key()
+
+        :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKey`,
+        :class:`~cryptography.hazmat.primitives.asymmetric.dsa.DSAPublicKey`,
+        :class:`~cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePublicKey`
+        or
+        :class:`~cryptography.hazmat.primitives.asymmetric.ed25519.Ed25519PublicKey`
+
+        The public key used to sign the certificate.
+
+    .. attribute:: signature
+
+        :type: bytes
+
+        The signature of the certificate. SSH signature serialization
+        formats do not always match the signature format used by
+        this library's other signature interfaces so if you need
+        to validate that the SSH certificate signature matches the
+        embedded signature key you should use
+        :meth:`check_cert_signature`.
+
```

Comment:
Given we have `check_cert_signature`, do users need this?

Reply:
If someone wanted to implement it themselves they'd need access to this and the unsigned body (which isn't currently exposed). We either need to expose both, or neither. I'm fine choosing neither until someone asks for it.

Reply:
I vote neither